internal/internal_workflow_testsuite.go

```diff
@@ -3229,47 +3268,61 @@ func (r *testNexusHandler) StartOperation(
 	m := &mockWrapper{
 		env:           r.env,
 		name:          service,
-		fn:            fn.Func.Interface(),
+		fn:            nil,
 		isWorkflow:    false,
 		dataConverter: r.env.dataConverter,
 	}
-	mockRet := m.getNexusMockReturn(ctx, operation, ptr.Elem().Interface(), r.opHandle.params.options)
+	mockRet := m.getNexusMockReturn(
+		ctx,
+		operation,
+		inputPtr.Elem().Interface(),
+		r.opHandle.params.options,
+	)
 	if mockRet != nil {
 		mockRetLen := len(mockRet)
 		if mockRetLen != 2 {
 			panic(fmt.Sprintf(
-				"mock of ExecuteNexusOperation has incorrect number of returns, expected 2, but actual is %d",
+				"mock of ExecuteNexusOperation has incorrect number of returns, expected 2, got %d",
 				mockRetLen,
 			))
 		}
 
-		// we already verified function either has 1 return value (error) or 2 return values (result, error)
-		var retErr error
-		mockErr := mockRet[mockRetLen-1] // last mock return must be error
-		if mockErr == nil {
-			retErr = nil
-		} else if err, ok := mockErr.(error); ok {
-			retErr = err
-		} else {
+		// we already verified function has 2 return values (result, error)
+		mockErr := mockRet[1] // last mock return must be error
+		if mockErr != nil {
+			if err, ok := mockErr.(error); ok {
+				return nil, err
+			}
 			panic(fmt.Sprintf(
-				"mock of ExecuteNexusOperation has incorrect return type, expected error, but actual is %T (%v)",
-				mockErr,
+				"mock of ExecuteNexusOperation has incorrect return type, expected error, got %T",
 				mockErr,
 			))
 		}
 
 		mockResult := mockRet[0]
-		ret, ok := mockResult.(nexus.HandlerStartOperationResult[any])
+		result, ok := mockResult.(nexus.HandlerStartOperationResult[any])
 		if mockResult != nil && !ok {
 			panic(fmt.Sprintf(
-				"mock of ExecuteNexusOperation has incorrect return type, expected nexus.HandlerStartOperationResult[T], but actual is %T (%v)",
-				mockResult,
+				"mock of ExecuteNexusOperation has incorrect return type, expected nexus.HandlerStartOperationResult[T], but actual is %T",
```

Comment:
nit: Is it a mock of `ExecuteNexusOperation`? I think of it as a mock of the operation handler.

Reply:
It's more like a mock of `ExecuteNexusOperation` because we are checking its input. Eg: we check the `NexusOperationOptions` which is not in the handler.